src/index.js

```diff
@@ -200,6 +204,87 @@ class IpfsRepo {
   exists (callback) {
     this.version.exists(callback)
   }
+
+  /**
+   * Get repo status.
+   *
+   * @param {Object}  options
+   * @param {Boolean} options.human
+   * @param {function(Error, Object)} callback
+   * @return {void}
+   */
+  stat (options, callback) {
+    if (typeof options === 'function') {
+      callback = options
+      options = {}
+    }
+
+    options = Object.assign({}, {human: false}, options)
+
+    parallel([
+      (cb) => cb(null, this.path),
+      (cb) => this.config.get('Datastore.StorageMax', (err, max) => {
+        if (err) {
+          cb(null, new Big(noLimit))
+        } else {
+          cb(null, new Big(max))
+        }
+      }),
+      (cb) => this.version.get(cb),
+      (cb) => waterfall([
+        (cb) => cb(null, {
+          repoSize: new Big(0)
+        }),
+        (stats, cb) => this.blocks.query({}, (err, list) => {
```

Comment:
the size fetching below can all happen in parallel, just need to sum at the end

Reply:
Done!